### PR TITLE
Feature/link cmd datasets to legacy datasets

### PR DESCRIPTION
**Pull request by jondewijones**
### What

Update legacy handler to handle filterable datasets and retrieve details using dataset client

### How to review

Pull branches:
https://github.com/ONSdigital/dp-frontend-renderer/pull/290
https://github.com/ONSdigital/sixteens/pull/157
https://github.com/ONSdigital/zebedee/pull/280

Enabled CMD through feature flags e.g. `DATASET_ROUTES_ENABLED=true` on router

Edit a `dataset_landing_page` either on disk or via Florence. Add a CMD datasets to the `relatedFilterableDatasets` field e.g:

```
"relatedFilterableDatasets": [
        {
            "uri": "/datasets/cpih01"
        },
        {
            "uri": "/datasets/ashe-table-7-hours"
        }
    ]
```

Then view the dataset landing page you have edited.

### Who can review

Anyone

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -424,6 +426,33 @@ func legacyLanding(w http.ResponseWriter, req *http.Request, zc ZebedeeClient, r
...
+		for _, relatedFilterableDataset := range dlp.RelatedFilterableDatasets {
+			wg.Add(1)
+			go func(ctx context.Context, dc DatasetClient, relatedFilterableDataset data.Related) {
+				defer wg.Done()
+				d, err := dc.GetByPath(ctx, relatedFilterableDataset.URI)
+				if err != nil {
+					// log error but continue to map data. any datasets that fail won't get mapped and won't be displayed on frontend
+					log.ErrorCtx(req.Context(), errors.WithMessage(err, "error fetching dataset details"), log.Data{
+						"dataset": relatedFilterableDataset.URI,
+					})
```
Comment (eldeal) [inline handlers/handlers.go]: trivial but might be slightly more idiomatic to have a return in this error condition so the 'happy path' is not nested in the else block

**Comment by dandwal** [comment]
Tests failing: 

> "handlers/handlers_test.go:80:31: cannot use mockDatasetClient (type *MockDatasetClient) as type DatasetClient in argument to testResponse:
> 	*MockDatasetClient does not implement DatasetClient (missing GetByPath method)"
> (FAIL	github.com/ONSdigital/dp-frontend-dataset-controller/handlers)

Likely need to update the tests

**Comment by dandwal** [comment]
Tests failing: 

> "handlers/handlers_test.go:80:31: cannot use mockDatasetClient (type *MockDatasetClient) as type DatasetClient in argument to testResponse:
> 	*MockDatasetClient does not implement DatasetClient (missing GetByPath method)"
> (FAIL	github.com/ONSdigital/dp-frontend-dataset-controller/handlers)

Likely need to update the tests

**Comment by dandwal** [comment]
Tests failing: 

> "handlers/handlers_test.go:80:31: cannot use mockDatasetClient (type *MockDatasetClient) as type DatasetClient in argument to testResponse:
> 	*MockDatasetClient does not implement DatasetClient (missing GetByPath method)"
> (FAIL	github.com/ONSdigital/dp-frontend-dataset-controller/handlers)

Likely need to update the tests

**Pull request merged**